profiler/src/main/java/com/splunk/opentelemetry/profiler/context/SpanContextualizer.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.profiler.context;
+
+import com.splunk.opentelemetry.profiler.events.ContextAttached;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import jdk.jfr.consumer.RecordedEvent;
+import jdk.jfr.consumer.RecordedThread;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Keeps track of span scope changes and, when applicable, can wrap the RecordedEvent with span
+ * context information. This class is not thread safe.
+ */
+public class SpanContextualizer {
+
+  private static final Logger logger = LoggerFactory.getLogger(SpanContextualizer.class);
+  private final long CANT_PARSE_THREAD_ID = Long.MIN_VALUE;
+  private final Pattern lineSplitter = Pattern.compile("\n");
+  private final ThreadContextTracker threadContextTracker = new ThreadContextTracker();
+
+  /**
+   * This updates the tracked context for a This must only be called with ContextAttached events.
+   */
+  public void updateContext(RecordedEvent event) {
+    if (event.getByte("direction") == ContextAttached.IN) {
```

Comment:
what a weird API. it's like a map of named bytes, or something?

Reply:
JFR is DEFINITELY a weird API, but in this particular case with this custom JFR event we get to choose the type.  I figured byte was compact and friendly. ..but am open to other ideas too.  Because there only two values, a boolean _could_ work but felt clunky.

Reply:
nah, this is fine. Just seemed weird to get a single byte, but have it keyed by 9 characters. :)